src/Abstractions/TaskName.cs

```diff
@@ -8,127 +8,162 @@ namespace Microsoft.DurableTask;
+  /// <param name="b">The second <see cref="TaskName"/> to compare.</param>
+  /// <returns><c>true</c> if the two <see cref="TaskName"/> objects are equal; otherwise <c>false</c>.</returns>
+  public static bool operator ==(TaskName a, TaskName b)
+  {
+    return a.Equals(b);
+  }
+
+  /// <summary>
+  /// Compares two <see cref="TaskName"/> objects for inequality.
+  /// </summary>
+  /// <param name="a">The first <see cref="TaskName"/> to compare.</param>
+  /// <param name="b">The second <see cref="TaskName"/> to compare.</param>
+  /// <returns><c>true</c> if the two <see cref="TaskName"/> objects are not equal; otherwise <c>false</c>.</returns>
+  public static bool operator !=(TaskName a, TaskName b)
+  {
+    return !a.Equals(b);
+  }
+
+  /// <summary>
+  /// Gets a value indicating whether to <see cref="TaskName"/> objects
+  /// are equal using value semantics.
+  /// </summary>
+  /// <param name="other">The other object to compare to.</param>
+  /// <returns><c>true</c> if the two objects are equal using value semantics; otherwise <c>false</c>.</returns>
+  public bool Equals(TaskName other)
+  {
+    return string.Equals(this.Name, other.Name, StringComparison.OrdinalIgnoreCase);
+  }
 
-    /// <summary>
-    /// Gets a value indicating whether to <see cref="TaskName"/> objects
-    /// are equal using value semantics.
-    /// </summary>
-    /// <param name="other">The other object to compare to.</param>
-    /// <returns><c>true</c> if the two objects are equal using value semantics; otherwise <c>false</c>.</returns>
-    public bool Equals(TaskName other)
+  /// <summary>
+  /// Gets a value indicating whether to <see cref="TaskName"/> objects
+  /// are equal using value semantics.
+  /// </summary>
+  /// <param name="obj">The other object to compare to.</param>
+  /// <returns><c>true</c> if the two objects are equal using value semantics; otherwise <c>false</c>.</returns>
+  public override bool Equals(object? obj)
+  {
+    if (obj is not TaskName other)
     {
-        return string.Equals(this.Name, other.Name, StringComparison.OrdinalIgnoreCase);
+      return false;
     }
 
-    /// <summary>
-    /// Gets a value indicating whether to <see cref="TaskName"/> objects
-    /// are equal using value semantics.
-    /// </summary>
-    /// <param name="obj">The other object to compare to.</param>
-    /// <returns><c>true</c> if the two objects are equal using value semantics; otherwise <c>false</c>.</returns>
-    public override bool Equals(object? obj)
-    {
-        if (obj is not TaskName other)
-        {
-            return false;
-        }
+    return this.Equals(other);
+  }
 
-        return this.Equals(other);
-    }
+  /// <summary>
+  /// Calculates a hash code value for the current <see cref="TaskName"/> instance.
+  /// </summary>
+  /// <returns>A 32-bit hash code value.</returns>
+  public override int GetHashCode()
+  {
+    return StringComparer.OrdinalIgnoreCase.GetHashCode(this.Name);
```

Comment:
We will need to update this to include `Version` in the hash code.

Reply:
Updated as your suggestion.